src/tools/miri/tests/fail/upvars/uninhabited-variant.rs

```diff
@@ -0,0 +1,31 @@
+// Motivated by #138961, this shows how invalid discriminants interact with
+// closure captures.
+#![feature(never_type)]
+
+#[repr(C)]
+#[allow(dead_code)]
+enum E {
+  V0, // discriminant: 0
+  V1, // 1
+  V2(!), // 2
+}
+
+fn main() {
+    assert_eq!(std::mem::size_of::<E>(), 4);
+
+    let val = 2u32;
+    let ptr = (&raw const val).cast::<E>();
+    let r = unsafe { &*ptr };
+    let f = || {
+        // After #138961, constructing the closure performs a reborrow of r.
```

Comment:
Since these are issue numbers in the rustc repo, but these tests will be mirrored to the Miri repo, please use full URLs instead of just issue numbers to avoid any ambiguity.